classes/dev/ComposerScript.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * @file classes/dev/ComposerScript.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ComposerScript
+ *
+ * @brief Custom composer scripts to run post installs/updates
+ */
+
+namespace PKP\dev;
+
+use Exception;
+
+class ComposerScript
+{
+	/**
+	 * A post-install-cmd custom composer script that
+	 * copies composer installs from repositories
+	 * to the correct/existing directories of the following dependencies:
+	 * jquery-ui and jquery validation
+	 */
+	public static function copyVendorAssets(): void
+	{   
+		function copyDir($src, $dst) {
+			$dir = opendir($src);
+			@mkdir($dst, 0755, true);
+			while (false !== ($file = readdir($dir))) {
+				if ($file != '.' && $file != '..') {
+					if (is_dir($src . '/' . $file)) {
+						copyDir($src . '/' . $file, $dst . '/' . $file);
+					} else {
+						copy($src . '/' . $file, $dst . '/' . $file);
+					}
+				}
+			}
+			closedir($dir);
+		}
```

Comment:
Hi @jonasraoni , thanks for checking this. I tried to implement your suggestion, but I'm running to lots of issues with importing the FileManager. When I run the `composer install`, I'm getting lots of "not found" errors so I did something like this when testing the use of FileManager:
```
                require_once './includes/bootstrap.inc.php';
		require_once './tools/core/Core.inc.php';
		require_once './classes/core/Core.inc.php';
		require_once './classes/config/Config.inc.php';
		require_once './classes/file/FileManager.inc.php';
		$fileManager = new \FileManager();
```
Finally, I run to an issue where the constant "INDEX_FILE_LOCATION" is undefined. As I check, this is defined on the app level of OJS/OMP/OPS.

I noticed a comment from this file [ComposerScript.php](https://github.com/pkp/pkp-lib/blob/main/classes/dev/ComposerScript.php#L28) from the main branch, that we did not use "Core::getBaseDir()" because the script is being called directly by composer and that "INDEX_FILE_LOCATION" is expected to be undefined. Would you be able to provide further guidance on this? Thanks!

Reply:
This should probably work:

```php
require_once './tools/bootstrap.inc.php';

import('lib.pkp.classes.file.FileManager');
$fileManager = new FileManager();
$fileManager->copyDir('source', 'destiny');
```

Reply:
This worked, thank you! I updated the script to use FileManager, please check. Thanks!

Reply:
Btw, I needed to update the FileManager file for 3.3 - specifically on the `mkdir` function:
![image](https://github.com/user-attachments/assets/79d09354-46be-49c1-9710-ace55b19d52f)

I added this because when I re-run the composer install (with already copied files), I get an error from php saying: 
![image](https://github.com/user-attachments/assets/c8f5f271-8c48-4a43-97ce-556009badcb5)

Weirdly, I don't get this error on 3.4 so I didn't change the FileManager file for 3.4.

Reply:
To not cause side-effects, the `return true;` should be probably modified to `return false;`.
Or modified to `return $this->setMode($dirPath, DIRECTORY_MODE_MASK);` to honor the expectations of the caller.

Reply:
I actually think it would be better not to use `FileManager`, since this is code that's run upon Composer dependency installation, long before there's a `config.inc.php` configuration file. `FileManager` expects the OJS bootstrapping process to be present, and it won't be.

Reply:
Thanks both, I updated the script to not use FileManager.